gnc/sub8_trajectory_generator/src/trajectory_generator.cc

```diff
@@ -4,13 +4,68 @@
 */
 #include <ros/ros.h>
 #include "sub8_state_space.h"
+#include "sub8_tgen_manager.h"
+#include "sub8_msgs/MotionPlan.h"
+
+namespace sub8 {
+namespace trajectory_generator {
+
+// Forward declaration for typedef
+class TGenNode;
+
+typedef boost::shared_ptr<TGenNode> TGenNodePtr;
+
+// Maintains a TGenManager object and defines callbacks for service
+// requests
+class TGenNode {
+ public:
+  TGenNode(int& planner_id) : _tgen(new Sub8TGenManager(planner_id)) {}
+  bool findTrajectory(sub8_msgs::MotionPlan::Request& req,
+                      sub8_msgs::MotionPlan::Response& resp) {
+    boost::shared_ptr<sub8_msgs::Waypoint> start_state_wpoint =
+        boost::make_shared<sub8_msgs::Waypoint>(req.start_state);
+    boost::shared_ptr<sub8_msgs::Waypoint> goal_state_wpoint =
+        boost::make_shared<sub8_msgs::Waypoint>(req.goal_state);
+
+    _tgen->setProblemDefinition(_tgen->waypointToState(start_state_wpoint),
+                                _tgen->waypointToState(goal_state_wpoint));
+
+    // What is the desired behavior here? If planning fails and no
+    // solution path is found, should we return false from the callback? 
+    // Or just set "resp.success" to false? If this happens, is it a 
+    // system-shutdown-worthy event? 
```

Comment:
Do we have a good idea of when trajectory generation fails? Will it fail under predictable conditions? Will we always be stationary when planning?